src/main/java/org/symphonyoss/symphony/entityjson/EntityJsonURIDownloader.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *
+ *
+ * Copyright 2017 Symphony Communication Services, LLC.
+ *
+ * Licensed to The Symphony Software Foundation (SSF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The SSF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.symphonyoss.symphony.entityjson;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+
+import com.github.fge.jsonschema.core.load.download.URIDownloader;
+
+/**
+ * URIDownloader which restricts schema downloads to known hosts.
+ * 
+ * @author Bruce Skingle
+ *
+ */
+public class EntityJsonURIDownloader implements URIDownloader
+{
+  private static final URIDownloader INSTANCE = new EntityJsonURIDownloader();
+
+  private EntityJsonURIDownloader()
+  {
+  }
+
+  public static URIDownloader getInstance()
+  {
+    return INSTANCE;
+  }
+
+  @Override
+  public InputStream fetch(final URI source) throws IOException
+  {
+    switch(source.getHost())
+    {
+      case "symphonyosf.github.io":
+      case "object.symphonyoss.org":
+      case "object.symphony.com":
+        return source.toURL().openStream();
```

Comment:
That's true but I'm implementing someone else's interface and there isn't anything I can do about it since it does not pass me a base URI. The interface does not explicitly specify that the URI is absolute but the purpose of the interface is to download the content and the default implementation in the library is

    @Override
    public InputStream fetch(final URI source)
        throws IOException
    {
        return source.toURL().openStream();
    }

So it is clearly implied that the URI will always be absolute. If it ever is not then yes, that exception will be thrown which would be the correct response.